### PR TITLE
Hash registry contract names

### LINKED ISSUE #527: Registry users SBAT not waste gas on hashing identifiers
### Expected Behavior

The hashing step when looking up things can easily be done off-chain. The Registry should have a function that looks up items based on a raw `bytes32` value rather than a string.

### Current Behavior

Users have to pass in a string that is then hashed on-chain.

### PR DESCRIPTION
**Pull request by bricewang**
### Description

Replaces registry ids in `UsingRegistry` with their packed `keccak256` hashes. Changes `getAddressFor` and `getAddressForOrDie` in `Registry` to take a `bytes32` hash. Changes tests using these functions to hash the name before passing it in.

### Tested

Tests refactored, no new ones needed.

### Related issues

- Fixes https://github.com/celo-org/celo-monorepo/issues/527
- Sister PR of https://github.com/celo-org/celo-blockchain/pull/418

### CODE REVIEW

**Comment by yerdua** [comment]
`contractkit` relies on being able to look up contract addresses from the registry, so this PR also necessitates changes to that code. `getAddressFor()` is called here: https://github.com/celo-org/celo-monorepo/blob/master/packages/contractkit/src/address-registry.ts#L21-L33

And this is where it has a mapping of contract name to registry key: https://github.com/celo-org/celo-monorepo/blob/master/packages/contractkit/src/base.ts#L3-L19

**Comment by m-chrzan** [comment]
I'm taking over this branch. Brice's changes look good to me, fixed up the thing Audrey pointed out. Waiting for another approval.

**Review by timmoreton — APPROVED** [review]
What is the plan to merge with this and the other PR? Take it you’ll merge both then raise a new PR to revert the circleci changes?

File: packages/protocol/contracts/common/Registry.sol
Comment (asaj) [inline packages/protocol/contracts/common/Registry.sol]: Since these are view calls should we continue to allow users to submit the string?

**Review by asaj — CHANGES_REQUESTED** [review]

File: packages/protocol/contracts/common/Registry.sol
Comment (mcortesi) [reply packages/protocol/contracts/common/Registry.sol]: I agree. more precisely, i think we can keep the old interface for everything. And thus have "retrocompatibility" :D

Comment (m-chrzan) [reply packages/protocol/contracts/common/Registry.sol]: While in this case having the string functions might be useful, I don't know that the reason for keeping them should be backwards compatibility. We don't need to remain backwards compatible with Alfajores, right? If anything, if we do find APIs that could be improved, we should completely replace them so that by mainnet we have as clean an interface as possible.

Comment (mcortesi) [reply packages/protocol/contracts/common/Registry.sol]: Sure, but what i propose is to do something like:

1. Make change, leave old api, mark it as deprectaded
2. Deploy
3. On a later PR, after all apps have already migrated to new contractkit/etc code that call new methods, remove deprecated methods

File: .circleci/config.yml
Comment (mcortesi) [inline .circleci/config.yml]: do we need this?

File: packages/protocol/test/common/integration.ts
Comment (mcortesi) [inline packages/protocol/test/common/integration.ts]: can't we use `Web3.utils.soliditySha3` and also avoid the `require`?

File: .circleci/config.yml
Comment (m-chrzan) [reply .circleci/config.yml]: This is temporary to get tests passing on this PR, making it mergeable. The idea would be to immediately after this and the geth PR get merged, to repoint back to master.

Comment (asaj) [reply .circleci/config.yml]: +1 to @mcortesi, if we merge the celo-blockchain PR first we can keep this pointed to master (or vice versa)

File: packages/protocol/contracts/common/Registry.sol
Comment (asaj) [inline packages/protocol/contracts/common/Registry.sol]: nit: consider `getAddressForStringOrDie` instead

File: packages/protocol/test/common/integration.ts
Comment (asaj) [reply packages/protocol/test/common/integration.ts]: +1, we should be able to use the `web3` provided to us by truffle

**Review by asaj — APPROVED** [review]

**Pull request merged**
